src/databricks/labs/ucx/progress/tables.py

```diff
@@ -54,15 +53,30 @@ def append_inventory_snapshot(self, snapshot: Iterable[Table]) -> None:
         # The mode is 'append'. This is documented as conflict-free.
         self._sql_backend.save_table(escape_sql_identifier(self.full_name), history_records, Historical, mode="append")
 
+    @cached_property
+    def _used_hive_tables(self) -> defaultdict[str, list[UsedTable]]:
```

Comment:
I just noticed this; the same review comments as on #3375 apply: enrichment data should be loaded within the scope of `.append_inventory_snapshot()` and not cached like this.

Reply:
I updated the code here, in the other PR it is more a refactor IMO as the cached_property is already used there, created a tech debt issue to resolve it after the PR